Source/Curves/ContinuousCurves/SMSplineCurveClass.f90

```diff
@@ -285,122 +309,158 @@ FUNCTION positionOnSplineCurveAt( self, t ) RESULT(x)
-   REAL(KIND=RP) FUNCTION seval(ndim,n,u,x,y,b,c,d) 
-!                                                                       
-!     .. scalar arguments ..                                            
-      REAL(KIND=RP) :: u 
-      integer n,ndim 
-!     ..                                                                
-!     .. array arguments ..                                             
-      REAL(KIND=RP) :: b(ndim),c(ndim),d(ndim),x(ndim),y(ndim) 
-!     ..                                                                
-!     .. local scalars ..                                               
-      REAL(KIND=RP) :: dx 
-      integer i,j,k 
-!     ..                                                                
-!     .. save statement ..                                              
-      save i 
-!     ..                                                                
-!     .. data statements ..                                             
-      data i/1/ 
-!     ..                                                                
-!                                                                       
-      if (i.ge.n) i = 1 
-      if (u.lt.x(i)) go to 100 
-      if (u.le.x(i+1)) go to 300 
-  100 continue 
-      i = 1 
-      j = n + 1 
-  200 continue 
-      k = (i+j)/2 
-      if (u.lt.x(k)) j = k 
-      if (u.ge.x(k)) i = k 
-      if (j.gt.i+1) go to 200 
-!                                                                       
-!     evaluate spline                                                   
-!                                                                       
-  300 continue 
-      dx = u - x(i) 
-      seval = y(i) + dx* (b(i)+dx* (c(i)+dx*d(i))) 
-!                                                                       
-      return 
-                                                                        
-   END FUNCTION seval                                          
+!
+!//////////////////////////////////////////////////////////////////////// 
+! 
+      SUBROUTINE ComputeSplineCoefs(n, x, y, b, c, d)
+!
+!     -------------------------------------------------------------------
+!     For derivations, see COMPUTER METHODS FOR MATHEMATICAL COMPUTATIONS
+!     by Forysyth, Malcom & Molder, Prentice-Hall 1977.
+!     -------------------------------------------------------------------
+!
+         IMPLICIT NONE
+!
+!        ---------
+!        Arguments
+!        ---------
+!
+         INTEGER, INTENT(IN)        :: n
+         REAL(KIND=RP), INTENT(IN)  :: x(n), y(n)
+         REAL(KIND=RP), INTENT(OUT) :: b(n), c(n), d(n)
+!
+!        ---------------
+!        Local Variables
+!        ---------------
+!
+         REAL(KIND=RP) :: h(n), alpha(n), beta(n)
+         INTEGER       :: i
+         
+         IF ( n < 4 )     THEN
+            b = 0.0_RP
+            c = 0.0_RP
+            d = 0.0_RP
+            RETURN 
+         END IF 
```

Comment:
I do not think this code can ever get triggered because if `N<4` an exception will get thrown before the `ComputeSplineCoefs` routine is called.

Reply:
That is true, but this does avoid (locally) any possibility of seg fault due to array bounds violations, in case it might get called accidentally without the check above.

Reply:
Aha, right. The function should still work correctly when something other than HOHQMesh uses it.